### ISSUE TITLE
Problems building/running the image in this tutorial

### ISSUE THREAD

**Issue by haroon-oxts** [MicrosoftDocs/visualstudio-docs#2814]
Thanks for putting this tutorial together, its exactly the type of thing I am trying to accomplish. Unfortunately I run into a number of issues when following it:

1. Error when downloading the required image during the build of this custom image:

```
PS C:\BuildTools&gt; docker build -t buildtools2017:latest -m 2GB .
Sending build context to Docker daemon  3.072kB
Step 1/6 : FROM microsoft/dotnet-framework:4.7.1
4.7.1: Pulling from microsoft/dotnet-framework
3889bb8d808b: Downloading [==================================================&gt;]  4.017GB/4.017GB
b479a92ce115: Downloading [==================================================&gt;]  1.553GB/1.553GB
420f6e4e21fd: Download complete
d280839cbd9a: Download complete
58c64f8af856: Download complete
2cd46d5789e9: Download complete
read tcp 192.168.79.106:52901-&gt;204.79.197.219:443: wsarecv: An existing connection was forcibly closed by the remote hos
t.
PS C:\BuildTools&gt;
```
It tends to do this quite often, restarting download attempts in a loop as they fail. Seems to be a network issue but the error message makes it seems like download server is closing the connection.

2. Once the package is downloaded, the build hangs on the call to the vs_buildtools call giving this output below. This happens for several ours and occasainnaly there 

```
Step 2/6 : SHELL ["cmd", "/S", "/C"]
 ---&gt; Running in 042d6049f7b8
Removing intermediate container 042d6049f7b8
 ---&gt; c3f5f8b95ccb
Step 3/6 : ADD https://aka.ms/vs/15/release/vs_buildtools.exe C:\TEMP\vs_buildtools.exe
Downloading [==================================================&gt;]  1.286MB/1.286MB
 ---&gt; ebce4fdf27a6
Step 4/6 : RUN C:\TEMP\vs_buildtools.exe --quiet --wait --norestart --nocache     --installPath C:\BuildTools     --all     --remove Microsoft.Visua
lStudio.Component.Windows10SDK.10240     --remove Microsoft.VisualStudio.Component.Windows10SDK.10586     --remove Microsoft.VisualStudio.Component.
Windows10SDK.14393     --remove Microsoft.VisualStudio.Component.Windows81SDK  || IF "%ERRORLEVEL%"=="3010" EXIT 0
 ---&gt; Running in 2de40259a00f
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1028\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\2052\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1055\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1046\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1042\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1029\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1036\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\3082\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1040\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1031\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1045\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1033\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1041\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\HelpFile\1049\help.html...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\vs_setup_bootstrapper.exe...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Microsoft.Diagnostics.Tracing.EventSource.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Microsoft.VisualStudio.RemoteControl.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Microsoft.VisualStudio.Setup.Common.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Microsoft.VisualStudio.Setup.Configuration.Interop.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Microsoft.VisualStudio.Setup.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Microsoft.VisualStudio.Setup.Download.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Microsoft.VisualStudio.Setup.Engine.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Microsoft.VisualStudio.Telemetry.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Microsoft.VisualStudio.Utilities.Internal.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\Newtonsoft.Json.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\zh-Hans\vs_setup_bootstrapper.resources.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\zh-Hant\vs_setup_bootstrapper.resources.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\cs\vs_setup_bootstrapper.resources.dll...
Preparing: C:\Users\ContainerAdministrator\AppData\Local\Temp\4a22d7c347855041ef06a9\vs_bootstrapper_d15\pt-BR\vs_setup_bootstrapper.resources.dll...
Prep
```
Some times this times out with an error message, but that takes multiples hours. I have not been able to get beyond this point on the machine were this happens. I feel like it might be a space issue as there is always 20 GB left over each time I try this.

3. On another machine the build works but I get the following error when running the image:

```
The system cannot find the path specified.
C:\Windows\system32&gt;
```

Powershell does not open and I can still see all the directories of my host PC, so it doesnt look like I'm in the container.

Any help is much appreciated.

PC information (from point number 2.):

> 
> Windows 10 on a surface book 
> 
> Docker version:
> Client: Docker Engine - Community
>  Version:           18.09.2
>  API version:       1.39
>  Go version:        go1.10.8
>  Git commit:        6247962
>  Built:             Sun Feb 10 04:12:31 2019
>  OS/Arch:           windows/amd64
>  Experimental:      false
> 
> Server: Docker Engine - Community
>  Engine:
>   Version:          18.09.2
>   API version:      1.39 (minimum version 1.24)
>   Go version:       go1.10.6
>   Git commit:       6247962
>   Built:            Sun Feb 10 04:28:48 2019
>   OS/Arch:          windows/amd64
>   Experimental:     false
> 
> Configuration in docker Daemon: 
> {
>   "registry-mirrors": [],
>   "insecure-registries": [],
>   "debug": true,
>   "experimental": false,
>   "storage-opts": [
>     "size=120GB"
>   ]
> }

  

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b91a4a43-5353-97cf-7532-7644aa3b497f
* Version Independent ID: 858d9a5c-72f3-6a85-9270-0a0dd9e0438b
* Content: [Install Visual Studio Build Tools into a container](https://docs.microsoft.com/en-us/visualstudio/install/build-tools-container?view=vs-2017#feedback)
* Content Source: [docs/install/build-tools-container.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/install/build-tools-container.md)
* Product: **visual-studio-windows**
* GitHub Login: @heaths
* Microsoft Alias: **tglee**

**Comment by heaths** [comment]
There's likely no hang. Nothing gets written to the console during the main installation (only the bootstrapper when extracting information). It can take several hours but you should notice the size changing. We install the full build tools as part of build validation and on pretty beefy hardware it takes 2-3 hours.

As for the networking issue, please see https://github.com/docker/for-win for support of Docker for Windows.

**Comment by apatel-oxts** [comment]
Hi, as a follow up to point 3 above, the docker file successfully builds completing all the steps. However when running the image, the system can not find the installed build tools. 

When I enter interactively into the container, I can see the Microsoft Visual Studio installer is inside the program files but no where in the container is the installed build tools. 

Any help on why the build tools are not being installed would be appreciated.

**Comment by heaths** [comment]
It's located wherever you told it to install. If you didn't supply an `--installPath` (highly recommended that you do), it defaults to somewhere under "%ProgramFiles(x86)%\Microsoft Visual Studio". If nothing but the "Installer" directory is there or the path you specified for the `--installPath` argument doesn't exist, we would need to see logs from the container. But please file a product bug via https://developercommunity.visualstudio.com.

**Comment by Mr-jiayunfei** [comment]
> Hi, as a follow up to point 3 above, the docker file successfully builds completing all the steps. However when running the image, the system can not find the installed build tools.
> 
> When I enter interactively into the container, I can see the Microsoft Visual Studio installer is inside the program files but no where in the container is the installed build tools.
> 
> Any help on why the build tools are not being installed would be appreciated.

I have a same problem,i nedd your help ,can you tell me how to fix this problem?( I can see the Microsoft Visual Studio installer is inside the program files but no where in the container is the installed build tools.)

**Comment by Simran-B** [comment]
https://docs.microsoft.com/en-us/visualstudio/install/build-tools-container?view=vs-2017#create-and-build-the-dockerfile was updated to now use `--all` anymore but only install a single workload using `--add`. Otherwise it will try to install >60GB of components and might even hit the storage-opts size or sandbox size limit of 120GB(?). There is no limit of 20GB anymore as mentioned in older issues as far as I can tell, and setting storage-opts shouldn't be necessary (I actually ran into issues when it was set in the daemon.json).

Not sure if it's related to https://github.com/microsoft/hcsshim/pull/718 in any way, but should you want to resize the VHD then the following instructions might help:

It is not possible to resize the disk image using the Docker Desktop Settings UI. If you would like to update the size of the disk image (for example, to 128 GB), run the following command in PowerShell:

    Resize-VHD -Path 'C:\ProgramData\DockerDesktop\vm-data\DockerDesktop.vhdx' -SizeBytes 128gb
